source/SH3/graphics/texture.cpp

```diff
@@ -4,44 +4,57 @@
  *
  *  @copyright 2016  Palm Studios
  *
- *
  *  @date 2-1-2017
  *
  *  @author Jesse Buhagiar
  */
 #include <SH3/graphics/texture.hpp>
 #include <SH3/system/log.hpp>
+#include <SH3/arc/vfile.hpp>
+
 
 using namespace sh3_graphics;
 
-std::string texture::load_error::message() const
+//TODO: Scale the texture and then
+int sh3_texture::Load(sh3_arc& mft, const std::string& filename)
 {
-    std::string error;
-    switch(result)
-    {
-    case load_result::SUCCESS:
-        error = "Success";
-        break;
-    case load_result::INVALID_CHUNK:
-        error = "Invalid chunk";
-        break;
-    case load_result::END_OF_FILE:
-        error = "End of file";
-        break;
-    }
-    return error;
-};
-
-texture::load_error sh3_graphics::texture::Load(const std::string& filename, sh3_arc& arc)
-{
-    load_error error;
+    sh3_texture_header header;
+    sh3_arc_vfile file(mft, filename);
+    sh3_arc_vfile::read_error e;
+
+    int ret = file.ReadData(&header, sizeof(sh3_texture_header), e); // Read in the header
+
+    if(ret != sizeof(sh3_texture_header))
+        Log(LogLevel::WARN, "sh3_texture::Load( ): Header only read %d bytes of data! It should have been %d", ret, sizeof(sh3_texture_header));
+
+    if(header.bpp == 8)
+        die("8-bit textures are currently unsupported!");
+
+    Log(LogLevel::INFO, "Texture info:\n\ttexWidth:%d\n\ttexHeight:%d\n\ttexSize:%d\n\tbpp:%d", header.texWidth, header.texHeight, header.texSize, header.bpp);
+
+    file.Seek(sizeof(sh3_texture_header) + header.dataOffset, std::ios_base::beg);
 
-    if(arc.LoadFile(filename, pixbuff) == ARC_FILE_NOT_FOUND)
-    {
-        die("sh3_graphics::texture::Load( ): Unable to find file %s!", filename.c_str());
-    }
+    data.resize(header.texSize);
+    ret = file.ReadData(&data[0], header.texSize, e);
 
-    error.set_error(load_result::END_OF_FILE);
+    if(ret != header.texSize)
+        Log(LogLevel::WARN, "sh3_texture::Load( ): Only read %d out of %d bytes!", ret, header.texSize);
 
-    return error;
+    // Now actually create a logical texture with OpenGL on the gpu
+    glGenTextures(1, &tex);
+
+    glBindTexture(GL_TEXTURE_2D, tex);
```

Comment:
This could use `Bind()` and later `Unbind()`. Though apart from being shorter I don't see much benefit.